src/main/java/cz/cvut/kbss/termit/service/business/SparqlExternalVocabularyService.java

```diff
@@ -0,0 +1,236 @@
+     * @return first imported Vocabulary
+     */
+    @Transactional
+    @PreAuthorize("@vocabularyAuthorizationService.canCreate()")
+    @Override
+    public Vocabulary importFromExternalUris(List<String> vocabularyIris) {
+        Vocabulary firstImportedVocabulary = null;
+
+        for (String vocabularyIri : vocabularyIris) {
+            LOG.trace("Starting import of external vocabulary {}", vocabularyIri);
+            InputStream newVocabulary = downloadExternalVocabulary(vocabularyIri);
+            if (newVocabulary != null) {
+                URI uri = URI.create(vocabularyIri);
+                Vocabulary vocabulary = repositoryService.importVocabulary(uri, RDFFormat.TURTLE.getDefaultMIMEType(), newVocabulary);
+
+                // add types
+                vocabulary.addType(cz.cvut.kbss.termit.util.Vocabulary.s_c_pouze_pro_cteni);
+                vocabulary.addType(cz.cvut.kbss.termit.util.Vocabulary.s_c_externi);
+
+                final AccessControlList acl = aclService.createFor(vocabulary);
+                vocabulary.setAcl(acl.getUri());
+
+                if (repositoryService.find(uri).isEmpty()) { // The vocabulary is new
+                    eventPublisher.publishEvent(new VocabularyCreatedEvent(this, vocabulary.getUri()));
+                }
+
+                LOG.trace("Vocabulary {} import was successful.", vocabularyIri);
+
+                if (firstImportedVocabulary == null) {
+                    firstImportedVocabulary = vocabulary;
+                }
+            }
+
+        }
+        
+        return firstImportedVocabulary;
+    }
+
+    private InputStream downloadExternalVocabulary(String vocabularyIri) {
+        InputStream vocabularyFile = null;
+        try {
+            SPARQLRepository sparqlRepo = initSparqlRepository();
+            try {
+                RepositoryConnection conn = sparqlRepo.getConnection();
+
+                String sparqlQuery = String.format(Utils.loadQuery(EXPORT_FULL_VOCABULARY_QUERY), vocabularyIri);
+                GraphQuery graphQuery = conn.prepareGraphQuery(sparqlQuery);
+
+                GraphQueryResult result = graphQuery.evaluate();
+                ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                Rio.write(result, outputStream, RDFFormat.TURTLE);
+                vocabularyFile = new ByteArrayInputStream(outputStream.toByteArray());
+            } catch (QueryEvaluationException ex) {
+                 LOG.error(ex.getMessage());
+            } finally {
+                sparqlRepo.shutDown();
+            }
+        } catch (RepositoryException ex) {
+             LOG.error(ex.getMessage());
+        }
+    
+        return vocabularyFile;
+
+    }
+    
+    @Scheduled(cron = "${termit.external.reloadCron:-}")
+    @Transactional
+    @Override
+    public void reloadExternalVocabularies() {
+        LOG.debug("Reloading vocabularies at " + Instant.now());
```

Comment:
Incorrect logging pattern (`Instant.now()` should be passed as second argument instead of string concatenation.